### PR TITLE
Add pins from 5.1.0 to 5.0.x

### PR DESCRIPTION
**Pull request by mwittgen**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mwittgen** [comment]
@conda-forge-admin, please rerender

**Review by erykoff — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -59,9 +59,10 @@ outputs:
         - compilers
         - cmake
         # Coverage 6.3 is breaking with NFS and multi-process testing.
-        - coverage >=3.6,!=6.3
+        # Coverage 7.0 is breaking with + in pathnames.
+        - coverage >=3.6,!=6.3,<7
```
Comment (ktlim) [inline recipe/meta.yaml:63]: This can be `!=7.0.0` as they did fix the problem we were seeing.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -151,9 +152,10 @@ outputs:
         - networkx
         - numexpr =2
         - {{ pin_compatible('numpy') }}
-        - pandas
+        - pandas <2
+        - parsl =1.2
```
Comment (ktlim) [inline recipe/meta.yaml:156]: I don't think we needed parsl for 5.0.0.  But we can include it.

**Review by ktlim — APPROVED** [review]
It's fine as is; a couple of things that could be changed.

**Pull request merged**